pkg/plugins/runtime/k8s/controllers/namespace_controller.go

```diff
@@ -58,28 +54,9 @@ func (r *NamespaceReconciler) Reconcile(req kube_ctrl.Request) (kube_ctrl.Result
 		return kube_ctrl.Result{}, err
 	}
 
-	if req.Name == r.SystemNamespace && !r.SkipDefaultMeshCreation {
-		// Fetch default Mesh instance
-		mesh := &mesh_k8s.Mesh{}
-		name := kube_types.NamespacedName{Name: core_model.DefaultMesh}
-		if err := r.Get(ctx, name, mesh); err != nil {
-			if kube_apierrs.IsNotFound(err) {
-				err := mesh_managers.CreateDefaultMesh(r.ResourceManager, r.DefaultMeshTemplate)
-				if err != nil {
-					log.Error(err, "unable to create default Mesh")
-				}
-				return kube_ctrl.Result{}, err
-			}
-			log.Error(err, "unable to fetch Mesh", "name", name)
-			return kube_ctrl.Result{}, err
-		}
-	}
-
-	if r.CNIEnabled {
-		if label, ok := ns.Labels["kuma.io/sidecar-injection"]; ok && label == "enabled" {
-			err := r.createOrUpdateNetworkAttachmentDefinition(req.Name)
-			return kube_ctrl.Result{}, err
-		}
+	if label, ok := ns.Labels["kuma.io/sidecar-injection"]; ok && label == "enabled" {
+		err := r.createOrUpdateNetworkAttachmentDefinition(req.Name)
```

Comment:
Probably this one also should be moved to the `defaults` component one day

Reply:
This is Kubernetes specific so I don't think that's the case.